src/components/Filter/__mocks__/mockFilterExample.js

```diff
@@ -114,8 +116,9 @@ export class MockFilterExample extends React.Component {
     } else {
       filterText += value;
     }
-    filterText += '\n';
-    this.setState({ filtersText: this.state.filtersText + filterText });
+
+    let activeFilters = [...this.state.activeFilters, { label: filterText }];
+    this.setState({ activeFilters: activeFilters });
   };
 
   selectFilterType(filterType) {
```

Comment:
I missed this last time... but it's best to try to combine calls to `setState` when possible... as every call to `setState` forces a re-evaluation of the virtual dom in `render`. Maybe something like this can work here:
```
      this.setState(prevState => {
        return {
          currentValue: '',
          currentFilterType: filterType,
          filterCategory:
            filterType.filterType === 'complex-select'
              ? undefined
              : prevState.filterCategory,
          categoryValue:
            filterType.filterType === 'complex-select'
              ? ''
              : prevState.categoryValue
        };
      });
```

Not a big deal... just more of an awareness thing ;)

Reply:
Thanks. Fixed.